lib/ansible/plugins/action/fetch.py

```diff
@@ -108,13 +100,16 @@ def run(self, tmp=None, task_vars=None):
             else:
                 source_local = source
 
-            dest = os.path.expanduser(dest)
+            # ensure we only use file name, avoid relative paths
+            abs_dest = os.path.abspath(dest)
+            abs_orig = os.path.abspath(original_dest)
+            if not abs_dest.startswith(abs_orig):
```

Comment:
I think we need a slightly different solution here. Consider the following:

```
abs_dest = '/foo/bar/bazqux'
abs_orig = '/foo/bar/baz'
abs_dest.startswith(abs_orig) == True
```

We could ensure they have trailing `/`, or we might be able to use something like:

```
if '..' in os.path.relpath(abs_dest, abs_orig).split(os.sep):
```

Reply:
or compare dirnames()